cranelift/codegen/src/isa/riscv64/inst.isle

```diff
@@ -2142,3 +2145,17 @@
   (lower_bmask $I128 $I128 val)
   (let ((res ValueRegs (lower_bmask $I64 $I128 val)))
     (value_regs (value_regs_get res 0) (value_regs_get res 0))))
+
+
+;;;; Helpers for physical registers ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(decl gen_mov_from_preg (PReg) Reg)
+
+(rule
+  (gen_mov_from_preg rm)
+  (let ((rd WritableReg (temp_writable_reg $I64))
+        (_ Unit (emit (MInst.MovFromPReg rd rm))))
+    rd))
+
+(decl px_reg (u8) PReg)
+(extern constructor px_reg px_reg)
```

Comment:
This generic constructor seems a little dangerous because it allows construction of `PReg` values for allocatable registers, which should not be usable; these will ideally be caught by an assertion (e.g. when handling moves) but it would be better to avoid constructing them in the first place. Could we expose specific constructors for the registers we want (e.g., `fp_reg`, `sp_reg`) here instead?